tests/optimize/test_objective.py

```diff
@@ -266,6 +281,581 @@ def test_directory(self):
             y["rdf"]["('1', '1')"], sim_ens.rdf["1", "1"].table.tolist()
         )
 
+
+class test_RelativeEntropyDirectAverage(unittest.TestCase):
```

Comment:
These look quite comprehensive! Only thing I noticed (sorry if I missed it): is there anything to check the calculation of the temperature when it is not given?

Reply:
Oops, good catch! I've added in a test that should cover this and the case where temp cannot be computed